spyder/widgets/sourcecode/codeeditor.py

```diff
@@ -1179,7 +1179,8 @@ def linenumberarea_paint_event(self, event):
         active_line_number = active_block.blockNumber() + 1
 
         def draw_pixmap(ytop, pixmap):
-            painter.drawPixmap(0, ytop + (font_height-pixmap.height()) / 2,
+            pixmap_height = pixmap.height() / pixmap.devicePixelRatio()

```

Comment:
I updated the logic to use the existing is_pyqt46 flag and restored the original behavior for PyQt4.  It's as of yet untested on PyQt4, as I don't have that installed right now.  Let me know if it works, or I can test it later.